doc/examples/edges/plot_edge_filter.py

```diff
@@ -88,3 +88,73 @@
 
 plt.tight_layout()
 plt.show()
+
+######################################################################
+# As in the example above, we illustrate the rotational invariance
+# of the filters in this example. We see an rotationally invariant image
+# along with the angle of its analytical gradient in the top row.
+# The other two rows contain the difference between the different gradient
+# approximations (Sobel, Prewitt, Scharr & Farid) and the analytical gradient
+# Farid & Simoncelli derivative filters are most rotationally invariant, but
+# require a 5x5 kernel for this advantage to unfold.
+#
+# .. [1] https://en.wikipedia.org/wiki/Sobel_operator#Alternative_operators
+#
+# .. [2] Farid, Hany, and Eero P. Simoncelli. "Differentiation of discrete
+#     multidimensional signals."
+#     IEEE Transactions on image processing 13.4 (2004): 496-508.
+#
+# .. [3] https://en.wikipedia.org/wiki/Image_derivatives
+# #Farid_and_Simoncelli_Derivatives
+
+
+x, y = np.mgrid[-10:10:255j, -10:10:255j]
+img = np.sin(x ** 2 + y ** 2)
+
+imgx = 2 * x * np.cos(x ** 2 + y ** 2)
+imgy = 2 * y * np.cos(x ** 2 + y ** 2)
+
+angle = lambda dx, dy: np.mod(np.arctan2(dy, dx), np.pi)
+true_angle = angle(imgx, imgy)
+
+angle_farid = angle(farid_h(img), farid_v(img))
+angle_sobel = angle(sobel_h(img), sobel_v(img))
+angle_scharr = angle(scharr_h(img), scharr_v(img))
+angle_prewitt = angle(prewitt_h(img), prewitt_v(img))
+
+diff_farid = np.abs(true_angle - angle_farid)
```

Comment:
```
def diff_angle(angle_1, angle_2):
      return np.minimum(np.pi - np.abs(angle_1 - angle_2),
                        np.abs(angle_1 - angle_2))
  
diff_farid = diff_angle(true_angle, angle_farid)
diff_sobel = diff_angle(true_angle, angle_sobel)
diff_scharr = diff_angle(true_angle, angle_scharr)
diff_prewitt = diff_angle(true_angle, angle_prewitt)
```

Reply:
Since angles are defined modulo pi, the difference must also be taken modulo pi. In the examples, some bright points close to pi were visible on the vertical axis, now they are all close to zero with this change.